ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/EstablishedPeers.hs

```diff
@@ -388,29 +376,31 @@ jobPromoteColdPeer PeerSelectionActions {
                       * 2 ^ (pred failCount `min` maxColdPeerRetryBackoff)
                       )
             bigLedgerPeersSet = PublicRootPeers.getBigLedgerPeers publicRootPeers
+
+            st' = st { knownPeers            = KnownPeers.setConnectTimes
+                                                 (Map.singleton
+                                                   peeraddr
+                                                   (delay `addTime` now))
+                                                 knownPeers',
+                       inProgressPromoteCold = Set.delete peeraddr
+                                                 (inProgressPromoteCold st),
+                       fuzzRng = fuzzRng'
+                     }
+            cs' = peerSelectionStateToCounters st'
```

Comment:
I think we might be duplicating some work. We compute the number of established or established big ledger peers here, and again in the governor loop. I haven't look closely elsewhere at this 'issue' but I suspect we do something like this in other monitoring actions. Maybe the decision record should be extended to include counters so that the main governor can reuse the thunk evaluated here?

Reply:
Yes, we could do that. But this PR is already large.  Let's create an issue for this.

Reply:
Done